Core/src/org/sleuthkit/autopsy/casemodule/ImageDSProcessor.java

```diff
@@ -296,10 +315,27 @@ public void run(String deviceId, String imagePath, String timeZone, boolean igno
      * @param callback             Callback to call when processing is done.
      */
     private void run(String deviceId, String imagePath, int sectorSize, String timeZone, boolean ignoreFatOrphanFiles, String md5, String sha1, String sha256, DataSourceProcessorProgressMonitor progressMonitor, DataSourceProcessorCallback callback) {
-        if (ingestStream == null) {
-            ingestStream = new DefaultIngestStream();
-        }
-        addImageTask = new AddImageTask(new AddImageTask.ImageDetails(deviceId, imagePath, sectorSize, timeZone, ignoreFatOrphanFiles, md5, sha1, sha256, null), 
+
+	// If the data source or ingest stream haven't been initialized yet, do it now.
+    if (ingestStream == null) {
+        ingestStream = new DefaultIngestStream();
+    }
+    if (image == null) {
+	    try {
+            image = SleuthkitJNI.addImageToDatabase(Case.getCurrentCase().getSleuthkitCase(),
+                new String[]{imagePath}, sectorSize,
+                timeZone, md5, sha1, sha256, deviceId);
+            } catch (TskCoreException ex) {
+            logger.log(Level.SEVERE, "Error adding data source with path " + imagePath + " to database", ex);
+            final List<String> errors = new ArrayList<>();
+            errors.add(ex.getMessage());
+            callback.done(DataSourceProcessorCallback.DataSourceProcessorResult.CRITICAL_ERRORS, errors, new ArrayList<>());
+            return;
+	    }
+	}
```

Comment:
Let's indulge in a little code duplication for clarity, since by the time you get to this method, you have to keep looking back at the callers.

We can also give this another name to ease the cognitive burden of the overloads. Something like doAddImageProcess() or addImage() or addFilesForImage() whatever.